doc/source/coding_style.rst

```diff
@@ -135,3 +135,173 @@ update any projects dependent on the API while still being treated as
 "first-class" users.  Note that due to the complexity of maintaining
 multiple "release branches" in a repository, the number of active
 release branches should be between one and three.
+
+Docstring Examples Best-Practice
+---------------------------------
+Defining docstring examples for methods and classes are extremely 
+useful. The examples give users an easy place to start when trying 
+out the API, by showing them exactly how to operate on a method or 
+class. By using ``doctest`` through ``pytest``, docstring examples can 
+also be used to perform regression testing to verify that the code is 
+executing as expected.
+
+Setting Up ``doctest``
+~~~~~~~~~~~~~~~~~~~~~~
+First, install ``pytest``:
+
+.. code::
+
+    pip install pytest
+
+Then, ``doctest`` can be run on any Python file by running:
+
+.. code::
+
+    pytest --doctest-modules file.py
+
+Here, ``doctest`` will search for any examples in the docstrings and will then
+execute them to verify that they function as written.
+
+Using ``pytest`` Fixtures
+~~~~~~~~~~~~~~~~~~~~~~~~~
+To define a set up sequence before the ``doctest`` run or before a given 
+module is tested, ``pytest`` fixtures can be used. Fixtures allow docstring 
+examples to access shared objects, so there is no need to repeat the set 
+up in each example.
```

Comment:
```suggestion
examples to access shared objects, so there is no need to repeat the setup
in each example.
```
I think using "setup" as a noun makes more sense here, but feel free to ignore this.